### PR TITLE
Feature/task 1

### PR DESCRIPTION
**Pull request by VlastimilM**
First review version

### CODE REVIEW

File: onboarding-task/src/components/List.jsx [hunk 1]
```diff
@@ -20,71 +20,72 @@ class List extends PureComponent {
...
+
     this.setState({ items: newItems });
   };
 
-  // TODO edit and save contain a lot of duplicate code, refactor
-  _edit = (index) => {
-    const newItems = this.state.items.slice();
-    newItems[index].editing = true;
+  _startEditing = (index) => {
+    const newItems = [...this.state.items];
```
Comment (Suzii) [inline onboarding-task/src/components/List.jsx]: This is a bit inconsistent with other methods, where you are using `slice()`.

File: onboarding-task/src/components/List.jsx [hunk 2]
```diff
@@ -106,8 +107,13 @@ class List extends PureComponent {
           <div className="col-sm-12 col-md-offset-2 col-md-8">
             <ListGroup>
               {listItems}
+              <ListGroupItem>
+                <NewListItem
+                  addFunction={this._addItem}
+                  key={generatePseudoUniqueID()}
```
Comment (Suzii) [inline onboarding-task/src/components/List.jsx]: Delete.

**Comment by Suzii** [comment]
ItemForm could have been named Create or Add, so that it is clear what it does. Like this it can be form for editnig as well.

**Comment by PetrSvirak** [comment]
> d405f1d KA-115 fixed points mentioned in first review

What a commit message, one knows exactly what happened there based solemnly on the `git log` :stuck_out_tongue_closed_eyes: → do read [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/) :wink:

**Pull request merged**